src/sdl2/surface.rs

```diff
@@ -13,18 +15,31 @@ use rwops::RWops;
 
 use sys::surface as ll;
 
-pub struct Surface<'a> {
+/// Holds a `SDL_Surface`
+///
+/// When the `SurfaceContext` is dropped, it frees the `SDL_Surface`
+///
+/// *INTERNAL USE ONLY*
+pub struct SurfaceContext<'a> {
     raw: *mut ll::SDL_Surface,
     _marker: PhantomData<&'a ()>
 }
 
-impl<'a> Drop for Surface<'a> {
+impl<'a> Drop for SurfaceContext<'a> {
     #[inline]
     fn drop(&mut self) {
         unsafe { ll::SDL_FreeSurface(self.raw); }
     }
 }
 
+/// Holds a `Rc<SurfaceContext>`.
+///
+/// Note: If a `Surface` goes out of scope but it cloned its context,
+/// then the `SDL_Surface` will not be free'd until there are no more references to the `SurfaceContext`.
+pub struct Surface<'a> {
```

Comment:
I am unsure what you meant by having the context and the canvas implement !Send and !Sync. 
Are those traits? I could not find any reference to them online.